src/middleware-set-current-time.js

```diff
@@ -12,12 +12,23 @@ videojs.use('*', (player) => {
     // level), this middleware will capture the action. For internal seeks (generated at
     // the tech level), we use a wrapped function so that we can handle it on our own
     // (specified elsewhere).
-    setCurrentTime: (time) => {
-      if (player.vhs && player.currentSource().src === player.vhs.source_.src) {
+    setCurrentTime(time) {
+      if (player.vhs &&
+          player.currentSource().src === player.vhs.source_.src) {
         player.vhs.setCurrentTime(time);
       }
 
       return time;
+    },
+
+    // Sync VHS after play requests.
+    // This specifically handles replay where the video element will
+    // seek to 0, skipping the setCurrentTime middleware, then play
+    play() {
+      if (player.vhs &&
+          player.currentSource().src === player.vhs.source_.src) {
+        player.vhs.setCurrentTime(player.currentTime());
```

Comment:
I think I would feel better if we specifically handled the case where `player.currentTime() === 0` instead of always doing this